test/integration/kubernetes_deploy_test.rb

```diff
@@ -232,8 +232,7 @@ def test_invalid_k8s_spec_that_is_valid_yaml_fails_fast_and_prints_template
       "Template validation failed",
       /Invalid template: ConfigMap-hello-cloud-configmap-data.*yml/,
       "> Error message:",
-      "error validating data: ValidationError(ConfigMap.metadata): \
-unknown field \"myKey\" in io.k8s.apimachinery.pkg.apis.meta.v1.ObjectMeta",
+      /error validating data: .*myKey.* v1.ObjectMeta/,
```

Comment:
My error message locally was just enough different to be annoying:
``` error: error validating "/var/folders/pk/7tjdz4295zl0t0_h8dlp99z40000gp/T/ConfigMap-hello-cloud-configmap-data20181107-23607-xe3tdb.yml": error validating data: found invalid field myKey for v1.ObjectMeta; if you choose to ignore these errors, turn validation off with --validate=false```

Reply:
Which k8s version is that from? I'm surprise our CI hasn't caught that possibility. The `v1.ObjectMeta` part isn't so important as the "unknown field myKey" / "invalid field myKey"

Reply:
``` ➜ k version                                                                                                                 
Client Version: version.Info{Major:"1", Minor:"7", GitVersion:"v1.7.8", GitCommit:"bc6162cc70b4a39a7f39391564e0dd0be60b39e9", GitTreeState:"clean", BuildDate:"2017-10-05T06:54:19Z", GoVersion:"go1.8.3", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.6", GitCommit:"a21fdbd78dde8f5447f5f6c331f7eb6f80bd684e", GitTreeState:"clean", BuildDate:"2018-07-26T10:04:08Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}```

Reply:
Can you try upgrading your client? We don't support 1.7 anymore.

Reply:
Yep that fixed it. I expect that error to be from the server not client.

Reply:
Yeah, dry-run is currently client-side. There's an alpha in 1.12 for server-side dry run: https://kubernetes.io/docs/reference/using-api/api-concepts/#dry-run.